### PR TITLE
Version Rails from 6.1.4.1 to 7.0.0

### PR DESCRIPTION
**Pull request by veganstraightedge**
- Upgrade Rails gem to `7.0.0`
- Run `bundle exec rails app:update` to update many config file changes
- Enable (nearly) all `new_framework_defaults_7_0.rb`
  - One left after deploying Rails 7 `Rails.application.config.active_support.cache_format_version = 7.0`
  - Then update `config.load_defaults 6.1` to `7.0` in `application.rb`

### CODE REVIEW

File: config/initializers/filter_parameter_logging.rb [hunk 1]
```diff
@@ -1,4 +1,6 @@
 # Be sure to restart your server when you modify this file.
 
 # Configure sensitive parameters which will be filtered from the log file.
-Rails.application.config.filter_parameters += %i[password username]
+Rails.application.config.filter_parameters += %i[
+  passw secret token _key crypt salt certificate otp ssn
```
Comment (just1602) [inline config/initializers/filter_parameter_logging.rb:5]: I'd keep password and username, tho

**Review by just1602 — APPROVED** [review]

File: config/initializers/filter_parameter_logging.rb (hunk 1 above)
Comment (veganstraightedge) [reply config/initializers/filter_parameter_logging.rb:5]: Yep. Great catch!

File: config/initializers/content_security_policy.rb [hunk 2]
```diff
@@ -4,24 +4,23 @@
...
-#   policy.object_src  :none
-#   policy.script_src  :self, :https
-#   policy.style_src   :self, :https
-#   # If you are using webpack-dev-server then specify webpack-dev-server host
-#   policy.connect_src :self, :https, "http://localhost:3035", "ws://localhost:3035" if Rails.env.development?
-
-#   # Specify URI for violation reports
-#   # policy.report_uri "/csp-violation-report-endpoint"
+# Rails.application.configure do
+#   config.content_security_policy do |policy|
```
Comment (just1602) [inline config/initializers/content_security_policy.rb:8]: Later it would be nice to activate CSP.

**Review by just1602 — APPROVED** [review]
Looks great to me !

File: config/initializers/permissions_policy.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+# information see https://developers.google.com/web/updates/2018/06/feature-policy
+#
+Rails.application.config.permissions_policy do |f|
+  f.camera      :none
+  f.gyroscope   :none
+  f.microphone  :none
+  f.usb         :none
+  f.fullscreen  '*'
+  f.payment     :self
+end
```
Comment (astronaut-wannabe) [inline config/initializers/permissions_policy.rb:11]: this is cool. i didnt know this was in rails 7

**Review by astronaut-wannabe — APPROVED** [review]
lgtm

File: config/initializers/permissions_policy.rb (hunk 3 above)
Comment (just1602) [reply config/initializers/permissions_policy.rb:11]: Yeah, I find it pretty cool too!

File: config/initializers/content_security_policy.rb (hunk 2 above)
Comment (veganstraightedge) [reply config/initializers/content_security_policy.rb:8]: Agreed. 👍🏻
You're welcome to dig into that if that's something that interests you.

File: config/initializers/permissions_policy.rb (hunk 3 above)
Comment (veganstraightedge) [reply config/initializers/permissions_policy.rb:11]: News to me too! I was like _SGTM!_.

**Pull request merged**